runtime/bootstrapper.ts

```diff
@@ -23,35 +23,36 @@ export class SkyAppBootstrapper {
   public static processBootstrapConfig(): Promise<any> {
     if (SkyAppBootstrapper.config && SkyAppBootstrapper.config.auth) {
       return BBAuth.getToken()
-        .then((token) => {
-          const url = this.getUrl();
+        .then(() => {
+          const currentUrl = this.getUrl();
 
           const params = new SkyAppRuntimeConfigParams(
-            url,
+            currentUrl,
             this.config.params
           );
 
-          const currentEnvId = params.get('envid');
-
-          return BBContextProvider.ensureContext({
-            envId: currentEnvId,
+          const ensureContextArgs: BBContextArgs = {
+            envId: params.get('envid'),
             envIdRequired: params.isRequired('envid'),
+            leId: params.get('leid'),
+            leIdRequired: params.isRequired('leid'),
             svcId: params.get('svcid'),
             svcIdRequired: params.isRequired('svcid'),
-            url: url
-          }).then(({ envId }) => {
-            // The context provider was able to provide an environment ID when none
-            // was supplied to the app; add it to the URL's query string and continue
-            // loading the app.  Downstream constructors of SkyAppRuntimeConfigParams
-            // will then pick up the environment ID from the query string.
-            if (!currentEnvId && envId) {
-              history.replaceState(
-                {},
-                '',
-                addQSParam(url, 'envid', envId)
-              );
-            }
-          });
+            url: currentUrl
+          };
+
+          return BBContextProvider.ensureContext(ensureContextArgs)
+            .then((args) => {
```

Comment:
Does `args` have a type?

Reply:
It does, but because `BBContextProvider.ensureContext()` has an explicit return type of `Promise<BBContextArgs>`, the type for the `args` parameter in the callback is implicitly typed `BBContextArgs`.  Therefore there's no need to specify the type in the code.